### PR TITLE
Enable E2E testing for v1.16 release cycle

### PR DESCRIPTION
**Pull request by willie-yao**
**What type of PR is this?**
/kind failing-test

**What this PR does / why we need it**:
This PR updates main E2E config for the v1.16 dev cycle.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

- [ ] cherry-pick candidate

**TODOs**:

- [ ] squashed commits
- [ ] includes documentation
- [ ] adds unit tests

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Review by mboersma — APPROVED** [review]
/lgtm

**Comment by willie-yao** [comment]
> failed to get component source YAML from URL: failed to get https://github.com/kubernetes-sigs/cluster-api-provider-azure/releases/download/v1.15.0/infrastructure-components.yaml: got status code 404

@mboersma @jackfrancis  I'm guessing this should be merged after the release is published?

**Comment by mboersma** [comment]
> this should be merged after the release is published?

I think it has to be to pass CI. It's in the "Post-Release Steps" of the releasing doc: https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/main/docs/book/src/developers/releasing.md#update-test-provider-versions-skip-for-patch-releases

**Review by mboersma — APPROVED** [review]
/lgtm

File: test/e2e/config/azure-dev.yaml [hunk 1]
```diff
@@ -83,29 +83,29 @@ providers:
   - name: azure
     type: InfrastructureProvider
     versions:
-    - name: v1.13.4 # latest patch of earliest minor in supported v1beta1 releases; this is used for v1beta1 old --> v1beta1 latest clusterctl upgrades test only.
-      value: https://github.com/kubernetes-sigs/cluster-api-provider-azure/releases/download/v1.13.4/infrastructure-components.yaml
+    - name: v1.14.3 # latest patch of earliest minor in supported v1beta1 releases; this is used for v1beta1 old --> v1beta1 latest clusterctl upgrades test only.
```
Comment (mboersma) [inline test/e2e/config/azure-dev.yaml:86]: Should we rev this to v1.14.4 now?

**Review by mboersma — APPROVED** [review]
/lgtm
/approve

**Review by jackfrancis — APPROVED** [review]
/lgtm
/approve

**Pull request merged**